anaSbitReadout.py

```diff
@@ -0,0 +1,376 @@
+    - 16 it's the max value allowed!
+    - In each run it will be filled from 0 up to chHitPerCluster
+    """
+    vfatN = array('i', [0])
+    chHitPerCluster = array('i', [0])
+    vfatCH = array('i', 16*[0])
+    strip = array('i', 16*[0])
+    sbitSize = array('i', [0])
+    L1Delay = array('i', [0])
+
+    outT.Branch('vfatN', vfatN, 'vfatN/I')
+    outT.Branch('chHitPerCluster', chHitPerCluster, 'chHitPerCluster/I')
+    outT.Branch('vfatCH', vfatCH, 'vfatCH[chHitPerCluster]/I')
+    outT.Branch('strip', strip, 'strip[chHitPerCluster]/I')
+    outT.Branch('sbitSize', sbitSize, 'sbitSize/I')
+    outT.Branch('L1Delay', L1Delay, 'L1Delay/I')
+
+    """
+    Defining both VFAT and iEta histos
+    """
+    # initializing vfat 1Dhisto
+    # While strip & ch branch are filled with arrays, histos are filled with one entries at a time
+
+    vfat_h_strip = ndict()
+    vfat_h_ch = ndict()
+    vfat_h_sbitSize = ndict()
+    vfat_h_delay = ndict()
+
+    for i in range(0, 24):
+        vfat_h_strip[i] = r.TH1F("Strips_vs_hit{0}".format(
+            i), "vfat{0}".format(i), 128, 0., 128.)
+        vfat_h_ch[i] = r.TH1F("Chann_vs_hit{0}".format(
+            i), "vfat{0}".format(i), 128, 0., 128.)
+        vfat_h_sbitSize[i] = r.TH1F("Sbitsize_vs_hit{0}".format(
+            i), "vfat{0} sbitSize".format(i), 8, 0., 8.)
+        vfat_h_delay[i] = r.TH1F("L1A_Sbit_delay{0}".format(
+            i), "vfat{0} L1A delay".format(i), 4096, 0., 4096.)
+        vfat_h_ch[i].SetXTitle("Chann Num")
+        vfat_h_strip[i].SetXTitle("Strip Num")
+        vfat_h_strip[i].SetFillColorAlpha(r.kBlue, 0.35)
+        vfat_h_ch[i].SetFillColorAlpha(r.kBlue, 0.35)
+
+    # initializing eta 1Dhisto
+    ieta_h_strip = ndict()
+    ieta_h_ch = ndict()
+    ieta_h_sbitSize = ndict()
+    ieta_h_delay = ndict()
+
+    for i in range(1, 9):
+        ieta_h_strip[i] = r.TH1F("eta_Strips_vs_hit{0}".format(
+            i), "i#eta = {0} | i#phi (1,2,3)".format(i), 384, 0., 384.)
+        ieta_h_ch[i] = r.TH1F("eta_Chann_vs_hit{0}".format(
+            i), "i#eta = {0} | i#phi (1,2,3)".format(i), 384, 0., 384.)
+        ieta_h_sbitSize[i] = r.TH1F("eta_Sbitsize_vs_hit{0}".format(
+            i), "i#eta = {0} sbitSize".format(i), 8, 0., 8.)
+        ieta_h_delay[i] = r.TH1F("eta_L1A_Sbit_delay{0}".format(
+            i), "i#eta = {0} L1A delay".format(i), 4096, 0., 4096.)
+        ieta_h_strip[i].SetFillColorAlpha(r.kBlue, 0.35)
+        ieta_h_ch[i].SetFillColorAlpha(r.kBlue, 0.35)
+        ieta_h_strip[i].SetXTitle("Strip num")
+        ieta_h_ch[i].SetXTitle("Chan num")
+
+    # initializing 2Dhisto
+    hh_ieta_strip = ndict()
+    hh_ieta_ch = ndict()
+    hh_ieta_strip[0] = r.TH2I(
+        'ieta_strip', 'Strips summary        (i#phi = 1,2,3);strip number;i#eta', 384, 0, 384, 8, 0.5, 8.5)
+    hh_ieta_ch[0] = r.TH2I(
+        'ieta_ch', 'Channels summary        (i#phi = 1,2,3);chan number;i#eta', 384, 0, 384, 8, 0.5, 8.5)
```

Comment:
Will sum of square weights behave correctly if these are `TH2I`? e.g. how do the error bars behave? If you have two entries in a bin is the error correctly `1.4...`?